tardis/tardis_portal/models/parameters.py

```diff
@@ -435,15 +438,53 @@ def set_value(self, value):
             self.numerical_value = float(value)
         elif self.name.isDateTime():
             # We convert the value string into datetime object.
-            # dateutil.parser detects and converts many date formats and is quite
-            # permissive in what it accepts (form validation and API input
-            # validation happens elsewhere and may be less permissive)
+            # dateutil.parser detects and converts many date formats and is
+            # quite permissive in what it accepts (form validation and API
+            # input validation happens elsewhere and may be less permissive)
             datevalue = dateutil.parser.parse(value)
             if settings.USE_TZ and is_naive(datevalue):
                 datevalue = make_aware(datevalue, LOCAL_TZ)
             elif not settings.USE_TZ and is_aware(datevalue):
                 datevalue = make_naive(datevalue, LOCAL_TZ)
             self.datetime_value = datevalue
+        elif self.name.isLink():
+            # Always store the raw value as a string, even if setting
+            # the GenericForeignKey via link_id/link_ct
+            self.string_value = unicode(value)
+
+            try:
+                # We detect experiment or dataset view URLs
+                # (eg, /experiment/view/12345 or /dataset/123)
+                # and extract values to populate link_ct and link_id. This
+                # covers two common cases, allowing LINK Parameters to be
+                # properly created via the REST API.
+
+                match = resolve(value)
+                if match.view_name == u'api_dispatch_detail':
+                    model_name = match.kwargs.get(u'resource_name', None)
+                    if model_name not in ('experiment', 'dataset'):
+                        model_name, pk = None, None
+                    else:
+                        pk = match.kwargs.get('pk', None)
+                elif match.view_name.endswith('view_experiment'):
+                    model_name = 'experiment'
+                    pk = match.kwargs.get('experiment_id')
+                elif match.view_name.endswith('view_dataset'):
+                    model_name = 'dataset'
+                    pk = match.kwargs.get('dataset_id')
+                else:
+                    model_name, pk = None, None
+
+                if pk is not None and model_name is not None:
+                    self.link_id = pk
+                    self.link_ct = ContentType.objects.get(
+                        app_label='tardis_portal',
+                        model=model_name.lower())
+            except (ValueError, IndexError):
```

Comment:
Also catch Resolver404 exception here, for the case where the link value doesn't resolve to any url.